src/main/java/hibernate/tournament/service/TournamentService.java

```diff
@@ -0,0 +1,75 @@
+package hibernate.tournament.service;
+
+import hibernate.tournament.model.Team;
+import hibernate.tournament.model.Tournament;
+import hibernate.tournament.repository.TeamRepository;
+import hibernate.tournament.repository.TournamentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class TournamentService extends GeneralService<Tournament, Integer>{
+
+    private TournamentRepository tournamentRepository;
+    private TeamRepository teamRepository;
```

Comment:
It is very bad approach to inject multiple repositories into layer that owns single entity.
High probability of circular dependency.